### ISSUE TITLE
TimeOnly property does not seem to be persisted to Saga Data

### ISSUE THREAD

**Issue by KCAndersen** [Particular/NServiceBus#6589]
Apologies if this issue is misplaced, but I couldn't find a better place to report it, nor anything on the internet regarding this, but it seems Sagas so not properly persist the new TimeOnly struct. 

```
public class OrderSagaState : ContainSagaData
{
    public Guid UniqueId { get; set; }

    public TimeOnly ExecutionTime { get; set; }
}
```
When populated, this results in the following data being saved for the saga:

```
{
  "UniqueId": "a3f37619-7e7a-4fe4-82ff-c4789242110c",
}
```

It is assigned to just fine at runtime; the data's simply not persisted and when that saga comes back the value is set to the default value.

When I change the type to Datetime, the data is persisted as expected.

**Comment by mauroservienti** [comment]
Hi @KCAndersen,

Which persistence package are you using? It seems like a serialization issue.

**Comment by KCAndersen** [comment]
We’re using MSSql Server and NewstonsoftSerializer. Your answer got me thinking maybe I needed to simply update my nugget packages; after doing so, TimeOnly is persisting as expected. Thanks for steering me in the right direction! I’m not positive which update did it, but for someone’s future reference, I updated:

NServiceBus from 7.7.3 to 7.8.8
NServiceBus.Newtonsoft.Json from 2.3.0 to 2.4.0
NServiceBus.Persistence.Sql from 6.5.1 to 6.6.0
